# Gold Price Analysis using Pandas

## This project aims to analyze historical gold prices using Pandas DataFrames. We will explore the dataset to understand the key features, perform initial data wrangling, and apply data filtering techniques to extract meaningful insights

### Data source:https://www.kaggle.com/datasets/cvergnolle/gold-price-and-relevant-metrics


In [1]:
import pandas as pd
pd.options.display.max_columns = None
pd.options.display.max_rows=None

## 	Load the CSV file containing historical gold prices into a Pandas DataFrame.

In [9]:
Gold_prices=pd.read_csv("Gold Price Prediction.csv").dropna(how="all")

In [3]:
Gold_prices.head(15)

Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
0    8/7/24             2405.87            2384.90      2385.83   
1    8/6/24             2442.74            2405.87      2384.90   
2    8/5/24             2447.17            2442.74      2405.87   
3    8/2/24             2447.23            2447.17      2442.74   
4    8/1/24             2411.09            2447.23      2447.17   
5   7/31/24             2382.47            2411.09      2447.23   
6   7/30/24             2387.11            2382.47      2411.09   
7   7/29/24             2365.37            2387.11      2382.47   
8   7/26/24             2400.12            2365.37      2387.11   
9   7/25/24             2409.86            2400.12      2365.37   
10  7/24/24             2397.05            2409.86      2400.12   
11  7/23/24             2400.95            2397.05      2409.86   
12  7/22/24             2442.71            2400.95      2397.05   
13  7/19/24             2462.44            2442.71      2400.95   
14  7/18/24             2469.65            2462.44      2442.71   

    Price Tomorrow  Price Change Tomorrow  Price Change Ten  Std Dev 10  \
0              NaN                    NaN               NaN   30.155078   
1          2385.83                   0.93               NaN   29.423936   
2          2384.90                 -20.97               NaN   28.341301   
3          2405.87                 -36.87               NaN   28.616661   
4          2442.74                  -4.43               NaN   26.084796   
5          2447.17                  -0.06               NaN   25.307192   
6          2447.23                  36.14               NaN   28.431801   
7          2411.09                  28.62               NaN   34.908633   
8          2382.47                  -4.64               NaN   33.472691   
9          2387.11                  21.74               NaN   31.991884   
10         2365.37                 -34.75            -14.29   26.120482   
11         2400.12                  -9.74            -24.96   29.774000   
12         2409.86                  12.81              8.82   34.370081   
13         2397.05                  -3.90             41.79   37.996575   
14         2400.95                 -41.76              4.46   38.405274   

    Twenty Moving Average  Fifty Day Moving Average  200 Day Moving Average  \
0               2414.7450                 2368.2948              2189.80780   
1               2414.1525                 2367.7916              2187.85630   
2               2413.2305                 2367.1584              2185.96345   
3               2411.0920                 2365.7250              2183.85640   
4               2408.5670                 2363.5624              2181.55785   
5               2404.1030                 2362.2836              2179.18645   
6               2399.7615                 2361.7958              2176.81845   
7               2395.7900                 2362.1302              2174.66915   
8               2393.3445                 2362.7596              2172.64085   
9               2390.3255                 2362.6238              2170.45925   
10              2387.0200                 2363.0558              2168.25615   
11              2383.0825                 2362.2280              2165.85250   
12              2379.1965                 2360.8238              2163.46460   
13              2375.3835                 2360.0924              2160.83315   
14              2373.4090                 2358.9954              2158.20535   

    Monthly Inflation Rate  EFFR Rate  Volume   Treasury Par Yield Month  \
0                      3.0       5.33       96                      5.50   
1                      3.0       5.33       89                      5.50   
2                      3.0       5.33       86                      5.52   
3                      3.0       5.33       89                      5.54   
4                      3.0       5.33       83                      5.55   
5           

In [6]:
Gold_prices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 676 entries, 0 to 675
Data columns (total 21 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   Date                                    676 non-null    object 
 1   Price 2 Days Prior                      676 non-null    float64
 2   Price 1 Day Prior                       676 non-null    float64
 3   Price Today                             676 non-null    float64
 4   Price Tomorrow                          675 non-null    float64
 5   Price Change Tomorrow                   675 non-null    float64
 6   Price Change Ten                        666 non-null    float64
 7   Std Dev 10                              667 non-null    float64
 8   Twenty Moving Average                   676 non-null    float64
 9   Fifty Day Moving Average                676 non-null    float64
 10  200 Day Moving Average                  676 non-null    float6

### By calling the info method on the dataset we see that our Dataset is comprised of 21 columns and 676 rows. We see that date is an object datatype. We will convert it to dates. Our Dataset has no mising values as we see from the Non-Null count


In [12]:
Gold_prices["Date"] = pd.to_datetime(Gold_prices["Date"], exact=True, cache=True, format="%m/%d/%y")
Gold_prices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 676 entries, 0 to 675
Data columns (total 21 columns):
 #   Column                                  Non-Null Count  Dtype         
---  ------                                  --------------  -----         
 0   Date                                    676 non-null    datetime64[ns]
 1   Price 2 Days Prior                      676 non-null    float64       
 2   Price 1 Day Prior                       676 non-null    float64       
 3   Price Today                             676 non-null    float64       
 4   Price Tomorrow                          675 non-null    float64       
 5   Price Change Tomorrow                   675 non-null    float64       
 6   Price Change Ten                        666 non-null    float64       
 7   Std Dev 10                              667 non-null    float64       
 8   Twenty Moving Average                   676 non-null    float64       
 9   Fifty Day Moving Average                676 non-null  

## Exploratory Data Analysis

### Average price change over the course of time

In [6]:
Gold_prices["Price Change Tomorrow"].mean()*100

86.15703703703704

### Sorting the dataset by the Price Today column from highest price to lowest

In [61]:
Gold_prices.sort_values(by="Price Today",ascending=False)


Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
16    7/16/24             2411.71            2423.87      2469.65   
15    7/17/24             2423.87            2469.65      2462.44   
5     7/31/24             2382.47            2411.09      2447.23   
4      8/1/24             2411.09            2447.23      2447.17   
3      8/2/24             2447.23            2447.17      2442.74   
..        ...                 ...                ...          ...   
457   11/3/22             1646.87            1633.88      1629.77   
484   9/27/22             1643.90            1627.95      1628.84   
485   9/26/22             1672.80            1643.90      1627.95   
468  10/19/22             1650.67            1652.42      1627.28   
467  10/20/22             1652.42            1627.28      1627.04   

     Price Tomorrow  Price Change Tomorrow  Price Change Ten  Std Dev 10  \
16          2462.44                  -7.21            -58.56   36.507686   
15          2442.71                 -19.73            -15.21   40.262424   
5           2447.17                  -0.06               NaN   25.307192   
4           2442.74                  -4.43               NaN   26.084796   
3           2405.87                 -36.87               NaN   28.616661   
..              ...                    ...               ...         ...   
457         1681.30                  51.53            130.16   12.817554   
484         1656.48                  27.64             35.14   22.084499   
485         1628.84                   0.89             42.34   22.078596   
468         1627.04                  -0.24              6.60   24.524710   
467         1657.60                  30.56              2.73   21.089534   

     Twenty Moving Average  Fifty Day Moving Average  200 Day Moving Average  \
16               2361.1695                 2353.4344              2152.29150   
15               2367.8050                 2356.3498              2155.29055   
5                2404.1030                 2362.2836              2179.18645   
4                2408.5670                 2363.5624              2181.55785   
3                2411.0920                 2365.7250              2183.85640   
..                     ...                       ...                     ...   
457              1652.3880                 1676.2888              1806.91730   
484              1686.3570                 1728.1662              1755.29250   
485              1691.0855                 1729.7896              1747.14830   
468              1667.6690                 1702.2660              1816.87090   
467              1665.3810                 1699.0230              1815.90030   

     Monthly Inflation Rate  EFFR Rate  Volume   Treasury Par Yield Month  \
16                      3.0       5.33       85                      5.48   
15                      3.0       5.33       80                      5.47   
5                       3.0       5.33       78                      5.49   
4                       3.0       5.33       83                      5.55   
3                       3.0       5.33       89                      5.54   
..                      ...        ...      ...                       ...   
457                     7.1       3.08       97                      3.75   
484                     8.2       2.33      111                      2.71   
485                     8.2       2.33      111                      2.73   
468                     7.7       3.08      104                      3.31   
467                     7.7       3.08       92                      3.58   

     Treasury Par Yield Two Year  Treasury Par Yield Curve Rates (10 Yr)  \
16                          4.43                                    4.17   
15                          4.42                                    4.16   
5                           4.29                                    4.09   
4                           4.16                                    3.99   
3

### How many times the 20-day Moving Average crosses over the 50-day Moving Average

In [18]:
Gold_prices.value_counts(Gold_prices["Twenty Moving Average"]>Gold_prices["Fifty Day Moving Average"])

True     380
False    296
Name: count, dtype: int64

### How many times the 20-day MA crosses over the 50-day Moving average during the year 2023. This information can give us all the potential buy signals for gold during this particular year. At first we are extracting the dataset and assign it to the Dataset_1 variable.

In [62]:
Twenty_day_MA_over_50_day_Ma=Gold_prices["Twenty Moving Average"]>Gold_prices["Fifty Day Moving Average"]
year_2023=Gold_prices["Date"].between("1/1/2023","12/31/2023")
Dataset_1=Gold_prices[Twenty_day_MA_over_50_day_Ma & year_2023]
Dataset_1                        

Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
135  1/29/24             2020.74            2018.57      2031.27   
136  1/26/24             2016.53            2020.74      2018.57   
137  1/25/24             2029.52            2016.53      2020.74   
138  1/24/24             2020.87            2029.52      2016.53   
139  1/23/24             2029.63            2020.87      2029.52   
..       ...                 ...                ...          ...   
671   1/7/22             1810.28            1791.61      1796.41   
672   1/6/22             1813.88            1810.28      1791.61   
673   1/5/22             1804.27            1813.88      1810.28   
674   1/4/22             1815.73            1804.27      1813.88   
675   1/3/22             1804.64            1829.05      1804.27   

     Price Tomorrow  Price Change Tomorrow  Price Change Ten  Std Dev 10  \
135         2036.55                   5.28            -12.33    6.892091   
136         2031.27                  12.70              5.93   11.993735   
137         2018.57                  -2.17             13.47   13.832766   
138         2020.74                   4.21             20.17   13.708441   
139         2016.53                 -12.99              5.80   12.975381   
..              ...                    ...               ...         ...   
671         1801.52                   5.11             38.17         NaN   
672         1796.41                   4.80             47.40         NaN   
673         1791.61                 -18.67             28.30         NaN   
674         1810.28                  -3.60              0.45         NaN   
675         1813.88                   9.61             14.95         NaN   

     Twenty Moving Average  Fifty Day Moving Average  200 Day Moving Average  \
135              2032.4705                 2032.4164              1965.75085   
136              2034.0935                 2031.4186              1965.57205   
137              2036.3100                 2030.6630              1965.39520   
138              2038.6255                 2029.9002              1965.31715   
139              2041.7725                 2028.7780              1965.20305   
..                     ...                       ...                     ...   
671               450.8225                  180.3290                45.08225   
672               361.0020                  144.4008                36.10020   
673               271.4215                  108.5686                27.14215   
674               180.9075                   72.3630                18.09075   
675                90.2135                   36.0854                 9.02135   

     Monthly Inflation Rate  EFFR Rate  Volume   Treasury Par Yield Month  \
135                     3.1       5.33       92                      5.53   
136                     3.1       5.33      100                      5.54   
137                     3.1       5.33      101                      5.54   
138                     3.1       5.33       96                      5.52   
139                     3.1       5.33       97                      5.53   
..                      ...        ...      ...                       ...   
671                     7.5       0.10       71                      0.05   
672                     7.5       0.10       73                      0.04   
673                     7.5       0.10       74                      0.05   
674                     7.5       0.10       74                      0.06   
675                     7.5       0.10       74                      0.05   

     Treasury Par Yield Two Year  Treasury Par Yield Curve Rates (10 Yr)  \
135                         4.29                                    4.08   
136                         4.34                                    4.15   
137                         4.28                                    4.14   
138                         4.34                                    4.18   
139         

### Below we see the the 20-day MA crosses over the 50-day MA 144 times during 2023.

In [42]:
Gold_prices.value_counts(Twenty_day_MA_over_50_day_Ma & year_2023)

False    532
True     144
Name: count, dtype: int64

### We now creating a column with the percentage difference between the 20-day MA and 50-day MA for the Dataset_1 above. This will allow us to extract some usefull metrics such as  the maximum and minimum percentage changes together with their respective dates 

In [78]:
Percentage_change_20MA_50MA=(Dataset_1["Twenty Moving Average"]-Dataset_1["Fifty Day Moving Average"])/Dataset_1["Fifty Day Moving Average"]*100
Dataset_1.head(20)


Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
135  1/29/24             2020.74            2018.57      2031.27   
136  1/26/24             2016.53            2020.74      2018.57   
137  1/25/24             2029.52            2016.53      2020.74   
138  1/24/24             2020.87            2029.52      2016.53   
139  1/23/24             2029.63            2020.87      2029.52   
140  1/22/24             2023.88            2029.63      2020.87   
141  1/19/24             2009.87            2023.88      2029.63   
142  1/18/24             2028.33            2009.87      2023.88   
143  1/17/24             2054.39            2028.33      2009.87   
144  1/16/24             2048.99            2054.39      2028.33   
145  1/15/24             2035.10            2048.99      2054.39   
146  1/12/24             2026.83            2035.10      2048.99   
147  1/11/24             2031.14            2026.83      2035.10   
148  1/10/24             2030.88            2031.14      2026.83   
149   1/9/24             2045.57            2030.88      2031.14   
150   1/8/24             2044.56            2045.57      2030.88   
151   1/5/24             2043.06            2044.56      2045.57   
152   1/4/24             2059.68            2043.06      2044.56   
153   1/3/24             2063.73            2059.68      2043.06   
154   1/2/24             2062.90            2063.73      2059.68   

     Price Tomorrow  Price Change Tomorrow  Price Change Ten  Std Dev 10  \
135         2036.55                   5.28            -12.33    6.892091   
136         2031.27                  12.70              5.93   11.993735   
137         2018.57                  -2.17             13.47   13.832766   
138         2020.74                   4.21             20.17   13.708441   
139         2016.53                 -12.99              5.80   12.975381   
140         2029.52                   8.65              4.28   12.968554   
141         2020.87                  -8.76             10.03   12.485414   
142         2029.63                   5.75             30.90   13.162025   
143         2023.88                  14.01             30.09   13.107444   
144         2009.87                 -18.46              8.22    9.614595   
145         2028.33                 -26.06            -23.12   10.825410   
146         2054.39                   5.40            -30.42   12.312095   
147         2048.99                  13.89            -14.36   13.769330   
148         2035.10                   8.27            -10.30   15.034467   
149         2026.83                  -4.31             -1.62   16.151507   
150         2031.14                   0.26            -10.01   14.667554   
151         2030.88                 -14.69            -15.94   11.695827   
152         2045.57                   1.01            -20.68   10.979506   
153         2044.56                   1.50            -33.19   10.037328   
154         2043.06                 -16.62            -31.35   12.396198   

     Twenty Moving Average  Fifty Day Moving Average  \
135              2032.4705                 2032.4164   
136              2034.0935                 2031.4186   
137              2036.3100                 2030.6630   
138              2038.6255                 2029.9002   
139              2041.7725                 2028.7780   
140              2043.6095                 2027.4390   
141              2045.3275                 2025.9520   
142              2046.4960                 2024.1192   
143              2047.9200                 2022.8256   
144              2049.0400                 2021.6462   
145              2049.6710                 2020.4592   
146              2048.3380                 2018.9074   
147              2046.8735                 2017.7800   
148              2046.9070                 2016.7786   
149              2047.2225                 2015.9448   
150              2044.7950                 2014.9982   
151              2042.4015    

In [77]:
max_percentage_change=Dataset_1.loc[Dataset_1["% Change 20-day MA & 50-day MA"].idxmax()]
max_percentage_change

Date                                        1/18/22
Price 2 Days Prior                          1817.43
Price 1 Day Prior                           1819.22
Price Today                                 1814.33
Price Tomorrow                              1838.58
Price Change Tomorrow                         24.25
Price Change Ten                             -13.56
Std Dev 10                                11.544873
Twenty Moving Average                     1086.8175
Fifty Day Moving Average                    434.727
% Change 20-day MA & 50-day MA                150.0
200 Day Moving Average                    108.68175
Monthly Inflation Rate                          7.5
EFFR Rate                                       0.1
Volume                                           71
Treasury Par Yield Month                       0.05
Treasury Par Yield Two Year                    1.06
Treasury Par Yield Curve Rates (10 Yr)         1.87
DXY                                           95.22
SP Open     

### As we can see there was a significant 20-day MA crossover above the 50-day MA. In the early 2022 there was a significant bullish movement of the gold price

In [79]:
min_percentage_change=Dataset_1.loc[Dataset_1["% Change 20-day MA & 50-day MA"].idxmin()]
min_percentage_change

Date                                         1/29/24
Price 2 Days Prior                           2020.74
Price 1 Day Prior                            2018.57
Price Today                                  2031.27
Price Tomorrow                               2036.55
Price Change Tomorrow                           5.28
Price Change Ten                              -12.33
Std Dev 10                                  6.892091
Twenty Moving Average                      2032.4705
Fifty Day Moving Average                   2032.4164
% Change 20-day MA & 50-day MA              0.002662
200 Day Moving Average                    1965.75085
Monthly Inflation Rate                           3.1
EFFR Rate                                       5.33
Volume                                            92
Treasury Par Yield Month                        5.53
Treasury Par Yield Two Year                     4.29
Treasury Par Yield Curve Rates (10 Yr)          4.08
DXY                                           

### At 29 January of 2024 the minimum percentage change between the two MAs has occured.

In [84]:
Dataset_1.sort_values(by=["% Change 20-day MA & 50-day MA"],ascending=False)

Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
664   1/18/22             1817.43            1819.22      1814.33   
657   1/27/22             1849.68            1822.01      1797.32   
670   1/10/22             1791.61            1796.41      1801.52   
669   1/11/22             1796.41            1801.52      1820.88   
666   1/14/22             1825.36            1821.16      1817.43   
675    1/3/22             1804.64            1829.05      1804.27   
674    1/4/22             1815.73            1804.27      1813.88   
672    1/6/22             1813.88            1810.28      1791.61   
668   1/12/22             1801.52            1820.88      1825.36   
667   1/13/22             1820.88            1825.36      1821.16   
658   1/26/22             1843.57            1849.68      1822.01   
659   1/25/22             1834.58            1843.57      1849.68   
661   1/21/22             1838.58            1839.01      1834.58   
656   1/28/22             1822.01            1797.32      1792.11   
660   1/24/22             1839.01            1834.58      1843.57   
662   1/20/22             1814.33            1838.58      1839.01   
663   1/19/22             1819.22            1814.33      1838.58   
665   1/17/22             1821.16            1817.43      1819.22   
671    1/7/22             1810.28            1791.61      1796.41   
673    1/5/22             1804.27            1813.88      1810.28   
655   1/31/22             1797.32            1792.11      1797.92   
394   1/31/23             1927.97            1922.21      1927.43   
395   1/30/23             1930.45            1927.97      1922.21   
396   1/27/23             1948.44            1930.45      1927.97   
397   1/26/23             1937.09            1948.44      1930.45   
398   1/25/23             1931.37            1937.09      1948.44   
436   12/2/22             1773.48            1802.34      1798.03   
423  12/21/22             1787.33            1817.28      1814.94   
430  12/12/22             1789.33            1797.35      1782.11   
399   1/24/23             1926.08            1931.37      1937.09   
424  12/20/22             1793.13            1787.33      1817.28   
429  12/13/22             1797.35            1782.11      1809.72   
428  12/14/22             1782.11            1809.72      1807.69   
422  12/22/22             1817.28            1814.94      1793.30   
425  12/19/22             1776.57            1793.13      1787.33   
426  12/16/22             1807.69            1776.57      1793.13   
427  12/15/22             1809.72            1807.69      1776.57   
421  12/23/22             1814.94            1793.30      1798.58   
420  12/26/22             1793.30            1798.58      1801.94   
419  12/27/22             1798.58            1801.94      1812.46   
437   12/1/22             1748.84            1773.48      1802.34   
400   1/23/23             1932.06            1926.08      1931.37   
418  12/28/22             1801.94            1812.46      1805.86   
401   1/20/23             1906.99            1932.06      1926.08   
417  12/29/22             1812.46            1805.86      1815.00   
438  11/30/22             1741.31            1748.84      1773.48   
402   1/19/23             1908.81            1906.99      1932.06   
416  12/30/22             1805.86            1815.00      1823.86   
415    1/2/23             1815.00            1823.86      1829.31   
414    1/3/23             1823.86            1829.31      1836.63   
413    1/4/23             1829.31            1836.63      1855.00   
403   1/18/23             1918.01            1908.81      1906.99   
189  11/14/23             1937.99            1946.52      1962.57   
412    1/5/23             1836.63            1855.00      1832.80   
188  11/15/23             1946.52            1962.57      1960.42   
404   1/17/23             1920.39            1918.01      1908.81   
405   1/16/23             1898.12            1920.39      1918.01   
190  11

### Extracting gold price data from specific time periods.

In [17]:
# The first half of 2024

First_Half_of_2024=Gold_prices[Gold_prices["Date"].between("01/01/24","06/30/24")]
First_Half_of_2024

Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
28  2024-06-28             2321.37            2299.26      2326.73   
29  2024-06-26             2332.14            2321.37      2299.26   
30  2024-06-25             2320.79            2332.14      2321.37   
31  2024-06-24             2361.46            2320.79      2332.14   
32  2024-06-21             2330.63            2361.46      2320.79   
33  2024-06-20             2329.73            2330.63      2361.46   
34  2024-06-19             2321.51            2329.73      2330.63   
35  2024-06-18             2333.02            2321.51      2329.73   
36  2024-06-17             2303.26            2333.02      2321.51   
37  2024-06-14             2322.23            2303.26      2333.02   
38  2024-06-13             2316.43            2322.23      2303.26   
39  2024-06-12             2311.13            2316.43      2322.23   
40  2024-06-11             2293.85            2311.13      2316.43   
41  2024-06-10             2375.74            2293.85      2311.13   
42  2024-06-07             2354.75            2375.74      2293.85   
43  2024-06-06             2327.09            2354.75      2375.74   
44  2024-06-05             2348.47            2327.09      2354.75   
45  2024-06-04             2327.36            2348.47      2327.09   
46  2024-06-03             2345.01            2327.36      2348.47   
47  2024-05-31             2338.59            2345.01      2327.36   
48  2024-05-30             2360.67            2338.59      2345.01   
49  2024-05-29             2353.24            2360.67      2338.59   
50  2024-05-28             2334.20            2353.24      2360.67   
51  2024-05-27             2334.61            2334.20      2353.24   
52  2024-05-24             2383.23            2334.61      2334.20   
53  2024-05-23             2422.84            2383.23      2334.61   
54  2024-05-22             2427.81            2422.84      2383.23   
55  2024-05-21             2413.94            2427.81      2422.84   
56  2024-05-20             2380.32            2413.94      2427.81   
57  2024-05-17             2386.97            2380.32      2413.94   
58  2024-05-16             2358.73            2386.97      2380.32   
59  2024-05-15             2339.65            2358.73      2386.97   
60  2024-05-14             2360.48            2339.65      2358.73   
61  2024-05-13             2346.10            2360.48      2339.65   
62  2024-05-10             2310.43            2346.10      2360.48   
63  2024-05-09             2316.67            2310.43      2346.10   
64  2024-05-08             2327.59            2316.67      2310.43   
65  2024-05-07             2302.57            2327.59      2316.67   
66  2024-05-06             2304.87            2302.57      2327.59   
67  2024-05-03             2322.67            2304.87      2302.57   
68  2024-05-02             2296.55            2322.67      2304.87   
69  2024-05-01             2334.46            2296.55      2322.67   
70  2024-04-30             2337.94            2334.46      2296.55   
71  2024-04-29             2330.80            2337.94      2334.46   
72  2024-04-26             2316.27            2330.80      2337.94   
73  2024-04-25             2323.65            2316.27      2330.80   
74  2024-04-24             2334.01            2323.65      2316.27   
75  2024-04-23             2390.86            2334.01      2323.65   
76  2024-04-22             2380.14            2390.86      2334.01   
77  2024-04-19             2367.91            2380.14      2390.86   
78  2024-04-18             2382.59            2367.91      2380.14   
79  2024-04-17             2381.97            2382.59      2367.91   
80  2024-04-16             2344.53            2381.97      2382.59   
81  2024-04-15             2376.89            2344.53      2381.97   
82  2024-04-12             2335.60            2376.89      2344.53   
83  2024-04-11             2354.86            2335.60      2376.89   
84  2024-04-10      

In [18]:
# First Half of 2023

First_Half_of_2023=Gold_prices[Gold_prices["Date"].between("01/01/23","06/30/23")]
First_Half_of_2023

Date  Price 2 Days Prior  Price 1 Day Prior  Price Today  \
286 2023-06-30             1909.84            1908.92      1919.54   
287 2023-06-29             1915.59            1909.84      1908.92   
288 2023-06-28             1922.56            1915.59      1909.84   
289 2023-06-27             1921.11            1922.56      1915.59   
290 2023-06-26             1914.58            1921.11      1922.56   
291 2023-06-23             1933.92            1914.58      1921.11   
292 2023-06-22             1937.34            1933.92      1914.58   
293 2023-06-21             1952.50            1937.34      1933.92   
294 2023-06-20             1957.84            1952.50      1937.34   
295 2023-06-19             1958.41            1957.84      1952.50   
296 2023-06-16             1945.31            1958.41      1957.84   
297 2023-06-15             1943.99            1945.31      1958.41   
298 2023-06-14             1960.03            1943.99      1945.31   
299 2023-06-13             1960.75            1960.03      1943.99   
300 2023-06-12             1965.48            1960.75      1960.03   
301 2023-06-09             1944.53            1965.48      1960.75   
302 2023-06-08             1963.36            1944.53      1965.48   
303 2023-06-07             1961.37            1963.36      1944.53   
304 2023-06-06             1948.20            1961.37      1963.36   
305 2023-06-05             1976.98            1948.20      1961.37   
306 2023-06-02             1966.47            1976.98      1948.20   
307 2023-06-01             1958.75            1966.47      1976.98   
308 2023-05-31             1941.70            1958.75      1966.47   
309 2023-05-30             1946.67            1941.70      1958.75   
310 2023-05-29             1940.54            1946.67      1941.70   
311 2023-05-26             1960.01            1940.54      1946.67   
312 2023-05-25             1975.17            1960.01      1940.54   
313 2023-05-24             1969.74            1975.17      1960.01   
314 2023-05-23             1977.87            1969.74      1975.17   
315 2023-05-22             1958.46            1977.87      1969.74   
316 2023-05-19             1982.76            1958.46      1977.87   
317 2023-05-18             1989.66            1982.76      1958.46   
318 2023-05-17             2014.49            1989.66      1982.76   
319 2023-05-16             2011.05            2014.49      1989.66   
320 2023-05-15             2016.33            2011.05      2014.49   
321 2023-05-12             2031.49            2016.33      2011.05   
322 2023-05-11             2036.38            2031.49      2016.33   
323 2023-05-10             2021.35            2036.38      2031.49   
324 2023-05-09             2016.87            2021.35      2036.38   
325 2023-05-08             2049.98            2016.87      2021.35   
326 2023-05-05             2052.04            2049.98      2016.87   
327 2023-05-04             2016.00            2052.04      2049.98   
328 2023-05-03             1981.31            2016.00      2052.04   
329 2023-05-02             1989.78            1981.31      2016.00   
330 2023-05-01             1987.09            1989.78      1981.31   
331 2023-04-28             1991.69            1987.09      1989.78   
332 2023-04-27             2000.22            1991.69      1987.09   
333 2023-04-26             1995.51            2000.22      1991.69   
334 2023-04-25             1983.20            1995.51      2000.22   
335 2023-04-24             2005.13            1983.20      1995.51   
336 2023-04-21             1993.71            2005.13      1983.20   
337 2023-04-20             2005.62            1993.71      2005.13   
338 2023-04-19             1993.99            2005.62      1993.71   
339 2023-04-18             2004.40            1993.99      2005.62   
340 2023-04-17             2039.97            2004.40      1993.99   
341 2023-04-14             2018.21            2039.97      2004.40   
342 2023-04-13      

### Comparing some descriptive statistics between the two financial semesters

In [33]:
print("The first semester of 2024 price average is" ,First_Half_of_2024["Price Today"].mean(),"while the first semester of 2023 price average is", First_Half_of_2023["Price Today"].mean())   

The first semester of 2024 price average is 2205.6635937499996 while the first semester of 2023 price average is 1934.6064615384614


In [38]:
# Extracting the data of the day where the standard deviation was at maximum. 
Gold_prices.loc[Gold_prices["Std Dev 10"].idxmax()]


Date                                      2023-03-20 00:00:00
Price 2 Days Prior                                    1921.45
Price 1 Day Prior                                     1988.11
Price Today                                           1976.63
Price Tomorrow                                        1942.95
Price Change Tomorrow                                  -33.68
Price Change Ten                                          8.3
Std Dev 10                                          62.183588
Twenty Moving Average                               1863.2665
Fifty Day Moving Average                            1881.9734
200 Day Moving Average                             1777.90415
Monthly Inflation Rate                                    5.0
EFFR Rate                                                4.57
Volume                                                     92
Treasury Par Yield Month                                 4.34
Treasury Par Yield Two Year                              3.92
Treasury

In [42]:
## Extracting the data of the day where the volume was at maximum
Gold_prices.loc[Gold_prices["Volume "].idxmax()]

Date                                      2023-06-06 00:00:00
Price 2 Days Prior                                     1948.2
Price 1 Day Prior                                     1961.37
Price Today                                           1963.36
Price Tomorrow                                        1944.53
Price Change Tomorrow                                  -18.83
Price Change Ten                                       -26.02
Std Dev 10                                          11.767095
Twenty Moving Average                               1974.5535
Fifty Day Moving Average                            1992.4924
200 Day Moving Average                             1839.66155
Monthly Inflation Rate                                    3.0
EFFR Rate                                                5.08
Volume                                                    144
Treasury Par Yield Month                                 5.15
Treasury Par Yield Two Year                              4.51
Treasury